src/web/mjs/engine/Connectors.mjs

```diff
@@ -24,16 +25,22 @@ export default class Connectors {
         ];
         let userPlugins = await this._loadPlugins('hakuneko://plugins/');
         let internalPlugins = await this._loadPlugins('hakuneko://cache/mjs/connectors/');
+        let favoritePlugins = Engine.Storage.downloadedConnectors;
 
         await this.register(systemPlugins);
         await this.register(userPlugins);
         await this.register(internalPlugins);
+        await this.registerfav(favoritePlugins);
     }
 
     get list() {
         return this._list;
     }
 
+    get favlist() {
+        return this._favlist;
```

Comment:
Just get the connector IDs of downloaded lists as suggested in `Storage.mjs` and then filter the full list based of the downloaded...
e.g.
```javascript
let downloaded = Storage.getDownloadedConnectorIDs();
return this._list.filter(connector => downloaded.includes(connector.id));
```